client/state/data-layer/wpcom/sites/comments/replies/new/index.js

```diff
@@ -4,20 +4,29 @@
 import { COMMENTS_WRITE } from 'state/action-types';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import {
+	hasParentCommentId,
 	dispatchNewCommentRequest,
 	updatePlaceholderComment,
 	handleWriteCommentFailure,
 } from 'state/data-layer/wpcom/sites/utils';
 
 export const writePostComment = ( { dispatch }, action ) => {
-	if ( ! action.parentCommentId ) {
+	if ( ! hasParentCommentId( action ) ) {
 		return;
 	}
 
 	const path = `/sites/${ action.siteId }/comments/${ action.parentCommentId }/replies/new`;
 	dispatchNewCommentRequest( dispatch, action, path );
 };
 
+const onSuccess = ( { dispatch }, action, ...args ) => ! hasParentCommentId( action )
+	? updatePlaceholderComment( { dispatch }, action, ...args )
+	: null;
+
+const onFailure = ( { dispatch }, action, ...args ) => ! hasParentCommentId( action )
+	? handleWriteCommentFailure( { dispatch }, action, ...args )
+	: null;
+
 export default {
-	[ COMMENTS_WRITE ]: [ dispatchRequest( writePostComment, updatePlaceholderComment, handleWriteCommentFailure ) ]
+	[ COMMENTS_WRITE ]: [ dispatchRequest( writePostComment, onSuccess, onFailure ) ]
```

Comment:
> what inspection are you looking for?

In this example we only want to trigger a certain set of handlers when `parentId` is available, and another set when it is not. Another action type will do the trick of course, though it does seem a bit like a waste since the actions will be very similar.

> we don't have to use dispatchRequest()

😄 this might be worth looking into @rodrigoi as another option